src/components/App.js

```diff
@@ -1,14 +1,165 @@
+    const baseURL = "http://www.omdbapi.com/";
+    const apiKey = "95869d44";
+    const page = this.state.page;
+    return `${baseURL}?apikey=${apiKey}&${typeOfSearch}=${search}&page=${page}&type=movie`
+}
+
+  fetchResults() {
+    fetch(this.createUrl('s', this.state.query)).
+      then(response => response.json()).
+      then(body => {
+        this.state.page===1?
+          (this.setState({results: body.Search})):
+          (this.setState({results: this.state.results.concat(body.Search)}))
+          this.setState({resultsLeft: body.totalResults-(this.state.page*10)});
+        
+      })
+  }
+
+
+//****************************************//
+// Receive search query from Search componenent
+
+
+  receiveQuery(query) {
+    this.setState({
+      query: query,
+      page: 1
+    },this.fetchResults)
+  }
+
+//****************************************//
+// Receive selected movie from Result component
+
+  receiveMovie(selectedMovie) {
+    this.setState({
+      selectedMovie:selectedMovie
+    })
+  }
+
+//****************************************//
+// Request to add another page of results (sent from Results component)
+
+  receiveMoreMovies() {
+    this.setState({
+      page:this.state.page+1
+    },this.fetchResults)
+
+  }
+
+//****************************************//
+// Add a movie to favourites object (App state)
+
+  receiveFavourite(result) {
+    const favObject = this.state.favouritesObject
+    if (favObject.hasOwnProperty(result.imdbID)) {
+      delete favObject[result.imdbID]
+    } else {
+      favObject[result.imdbID] = result
+    }
+    this.setState({
+      favouritesObject:favObject
+    },localStorage.setItem('reactFavourites', JSON.stringify(this.state.favouritesObject)))
+
+    this.updateFavouritesArray();
+  }
+
+//****************************************//
+// Create an array of favourites from favourites object. This is used to update the App/result state which will then show favourites instead of search
+
+  updateFavouritesArray() {
+    console.log(this.state.favouritesObject)
+    this.state.favouritesObject.hasOwnProperty('test')?delete this.state.favouritesObject.test:null
+    const favouritesKeys = Object.keys(this.state.favouritesObject);
+    this.state.favouritesArray = favouritesKeys.map(key => this.state.favouritesObject[key])
+    console.log(this.state.favouritesArray);
+    this.setState({
+      favouritesLength:this.state.favouritesArray.length
+    })
+  }
+
+//****************************************//
+// By setting the state of result to the favourites array favourites will now be shown instead of search results
+
+  showFavourites() {
+    this.setState({
+      results:this.state.favouritesArray,
+      resultsLeft:0
+    })
   }
 
   render(){
     return (
-      <div>
-        React cinema app
+      <div className="app">
+        <div className="top">
+        <p className="top__favourites" onClick={this.showFavourites}>Favourites ({this.state.favouritesLength})</p>
```

Comment:
rather than storing `favouritesLength` in state, it would be better to use `Object.values(this.state.favouritesObject).length`. That will ensure that you always working from the same source of truth rather than having to recalculate favouritesLength.